### ISSUE TITLE
ngbTooltip and ngbModal interference on open/close of Modal

### ISSUE THREAD

**Issue by BenniG82** [ng-bootstrap/ng-bootstrap#3517]
### Bug description:
If you are using **ngbTooltip** on a element that opens a **ngbModal**, the tooltip disappears as soon as you click (and the modal opens) => good.
But as soon as you close the modal, the tooltip reappears and won't disappear until you enter/leave the tooltipped element or click anywhere.

### Link to minimally-working StackBlitz that reproduces the issue:

https://stackblitz.com/edit/angular-fddawz

### Versions of Angular, ng-bootstrap and Bootstrap:
Angular: 8.2.4
ng-bootstrap: 5.1.4
Bootstrap: 4.3.1

### Additional Note ###
I think the current behaviour might be useful in some scenarios but for the majority of use cases the ngbTooltip should remain closed after a ngbModal Dialog has been closed, so maybe a configuration property might be nice.

**Comment by BenniG82** [comment]
I just played around some more and went a bit through your code. This behaviour is caused by the button losing and regaining focus as soon as the modal is opened and then closed.

So setting the ngbTooltip triggers to _hover_ helps:
```
<button 
  class="btn btn-lg btn-outline-primary" 
  ngbTooltip="Tooltip"
  triggers="hover"
  (click)="open(content)"
>Launch demo modal</button>
```
So I think this is a feature, but it feels a bit odd to me. 

For us keyboard-focus change is not really relevant, so I can go with ```triggers="hover"```.
Maybe one of you wise folks has an idea if this should be changed or not.

**Comment by peterblazejewicz** [comment]
@BenniG82 
This is a feature - restoring the focus to the last element before the dialog appeared, see relevant paragraph here for example:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role#Focus_management

As the `document.activeElement` requires interactive element, one can just simply disable the trigger ui element when the dialog opens, and enable this after the dialog closes, as this should set the 'activeElement' to null, based on your snips:
```ts
this.modalService.open(content, {ariaLabelledBy: 'modal-basic-title'}).result.then((result) => {
      this.closeResult = `Closed with: ${result}`;
      this.disabled = false;
    }, (reason) => {
      this.closeResult = `Dismissed ${this.getDismissReason(reason)}`;
      this.disabled = false;
    });
    this.disabled = true;
```
```html
<button 
  class="btn btn-lg btn-outline-primary" 
  ngbTooltip="Tooltip"
  [disabled]="disabled"
  (click)="open(content)"
>Launch demo modal</button>
```

**Comment by BenniG82** [comment]
Thanks @peterblazejewicz for your answer.
Do you think making this behavior configurable may be an option? (See the last paragraph in the [answer to myself](https://github.com/ng-bootstrap/ng-bootstrap/issues/3517#issuecomment-566026270)). Something like:
```ts
this.modalService.open(content, {restoreFocusOnClose: false}).
```

**Comment by peterblazejewicz** [comment]
Yes, there could be some usel-case that asks for that support. But I think it wil be a minor priority.
I'd advise to rephrase issue title to cleary state what you're asking for to have.

**Comment by AccessViolator** [comment]
Bump. This is annoying.

**Comment by Andrew3005** [comment]
@peterblazejewicz hi, any updates how I can avoid focus on button after closing modal?
